Makefile

```diff
@@ -6,7 +8,8 @@ check-syntax: src
 check: src build.xml
 	ant
 
-dist/: src build.xml build.txt
+dist/: src build.xml build.txt $(SOURCES)
+	ant clean
```

Comment:
This part actually is wrong (it should only rebuild if there actually were changes — fixed now).

I assume that you also mean including a Makefile at all, but since Makefiles are the one cross-language standard for defining how to build a program, I disagree on this. I wish we had Makefiles in fred to drive gradle, so people who are not used to gradle would not have to hunt for the compile commands (when I wanted to do quick experiments in next that lost me much more time than I’m comfortable losing to a tool which should only turn a small change into something I can test and run). A plain Makefile would make switching build systems much less painful: just swap a line executing ant with a line executing gradle, and all which commandline users notice would be that the tests are faster and the build output cleaner. But I won’t press that philosophical discussion.